libgalois/include/galois/worklists/AdaptiveObim.h

```diff
@@ -357,63 +408,48 @@ struct AdaptiveOrderedByIntegerMetric : private boost::noncopyable {
         else
           delta = 0;
 
+        p.cleanup();
         for (unsigned i = 1; i < runtime::activeThreads; ++i) {
-          while (current.getRemote(i)->lock.try_lock())
+          while (!data.getRemote(i)->lock.try_lock())
             ;
+          data.getRemote(i)->cleanup();
+          data.getRemote(i)->lock.unlock();
         }
-
-        for (unsigned i = 0; i < runtime::activeThreads; ++i) {
-
-          current.getRemote(i)->sinceLastFix          = 0;
-          current.getRemote(i)->slowPopsLastPeriod    = 0;
-          current.getRemote(i)->pushesLastPeriod      = 0;
-          current.getRemote(i)->priosLastPeriod       = 0;
-          current.getRemote(i)->maxPrioDiffLastPeriod = 0;
-
-          current.getRemote(i)->minPrio = std::numeric_limits<Index>::max();
-          current.getRemote(i)->maxPrio = std::numeric_limits<Index>::min();
-        }
-
-        for (unsigned i = 1; i < runtime::activeThreads; ++i) {
-          current.getRemote(i)->lock.unlock();
-        }
-        p.ctr++;
+        p.stats.numUmerges++;
       }
       p.popsFromSameQ = 0;
     }
-#endif
-    p.popsFromSameQ = 0;
+    // p.popsFromSameQ = 0;
 
-    updateLocal(p);
     bool localLeader = substrate::ThreadPool::isLeader();
+    deltaIndex msS(this->earliest, 0);
+
+    updateLocal(p);
 
-    deltaIndex msS;
-    msS.k = std::numeric_limits<Index>::min();
```

Comment:
> (quoted from comments on L294 of the last commit)
This issue was present in the old version too, but we should use the max and min from the comparator here and not rely on the numeric_limits. A related question here though: does this scheduler even support prioritizing higher integer priorities instead of lower ones? Now that I think about it some more that may throw off some of the arithmetic elsewhere. Fortunately we're not relying on this feature right now, so it's not something we have to fix immediately.

Rather, the numeric min here is problematic (but fixed now). If you look back to the initial value of min/max (earliest/identity) from the comparator, the original code wrongly made them reversed (also fixed now).

Reply:
Awesome. Thanks for catching that!